src/wallet/wallet.h

```diff
@@ -576,6 +576,7 @@ class CWalletTx
 
     bool InMempool() const;
     bool IsTrusted(interfaces::Chain::Lock& locked_chain) const;
+    bool IsTrusted(interfaces::Chain::Lock& locked_chain, std::set<uint256>& trusted_parents) const;
```

Comment:
nit, should be private?

Reply:
Actually I'd rather just get rid of the other function as usually you'd want to share the cache across IsTrusted calls...

Slightly more agressive change but maybe it's ok